src/cmd-buildextend-azure

```diff
@@ -111,41 +122,40 @@ def cli():
         default=os.environ.get("AZURE_LOCATION", "westus"))
     parser.add_argument(
         '--profile', help='Path to Azure profile',
-        required=True, default=os.environ.get('AZURE_PROFILE'))
+        default=os.environ.get('AZURE_PROFILE'))
     parser.add_argument(
-        '--resource-group', help='Resource group', required=True,
+        '--resource-group', help='Resource group',
         default=os.environ.get('AZURE_RESOURCE_GROUP'))
     parser.add_argument(
-        '--storage-account', help='Storage account', required=True,
+        '--storage-account', help='Storage account',
         default=os.environ.get('AZURE_STORAGE_ACCOUNT'))
     parser.add_argument("--log-level", help="ore log level")
+    parser.add_argument("--upload", action='store_true', default=False,
+                        help="Upload to Azure")
     args = parser.parse_args()
 
-    # Argument checks for environment strings that are required
-    arg_exp_str = "parameter '--{}' or envVar '{}' must be defined"
-    if args.auth is None:
-        raise Exception(arg_exp_str.format('auth', 'AZURE_AUTH'))
-    if args.container is None:
-        raise Exception(arg_exp_str.format('container', 'AZURE_CONTAINER'))
-    if args.profile is None:
-        raise Exception(arg_exp_str.format('profile', 'AZURE_PROFILE'))
-    if args.resource_group is None:
-        raise Exception(arg_exp_str.format('resource_group', 'AZURE_RESOURCE_GROUP'))
-    if args.storage_account is None:
-        raise Exception(arg_exp_str.format('storage_account', 'AZURE_STORAGE_ACCOUNT'))
+    if args.upload:
+        # Argument checks for environment strings that are required
+        arg_exp_str = "parameter '--{}' or envVar '{}' must be defined when uploading"
+        if args.auth is None:
+            raise Exception(arg_exp_str.format('auth', 'AZURE_AUTH'))
+        if args.container is None:
+            raise Exception(arg_exp_str.format('container', 'AZURE_CONTAINER'))
+        if args.profile is None:
+            raise Exception(arg_exp_str.format('profile', 'AZURE_PROFILE'))
+        if args.resource_group is None:
+            raise Exception(arg_exp_str.format('resource_group', 'AZURE_RESOURCE_GROUP'))
+        if args.storage_account is None:
+            raise Exception(arg_exp_str.format('storage_account', 'AZURE_STORAGE_ACCOUNT'))
```

Comment:
Could at least shorten this (and the other instances) to:

```python
for attr in ['auth', 'container', 'profile', ...]:
  if getattr(args, attr) is None:
    raise Exception(f"parameter ...")
```

Reply:
fixed in followup commit